src/Checkbox/index.js

```diff
@@ -72,59 +82,100 @@ export type CheckboxPropsT = {
 /**
  * A Checkbox component
  */
-export const Checkbox = withMDCToggle({
-  mdcConstructor: MDCCheckbox
-})(
-  class extends React.Component<CheckboxPropsT> {
-    static displayName = 'Checkbox';
-
-    render() {
-      const {
-        label = '',
-        id,
-        children,
-        checked,
-        apiRef,
-        indeterminate,
-        mdcElementRef,
-        generatedId,
-        ...rest
-      } = this.props;
-      const labelId = id || generatedId;
-      const checkedProp = checked !== undefined ? { checked } : {};
-      const classes = classNames({ 'mdc-checkbox--disabled': rest.disabled });
-
-      const checkbox = (
-        <CheckboxRoot elementRef={mdcElementRef} className={classes}>
-          <CheckboxNativeControl id={labelId} {...checkedProp} {...rest} />
-          <CheckboxBackground>
-            <CheckboxCheckmark>
-              <CheckboxCheckmarkPath />
-            </CheckboxCheckmark>
-            <CheckboxMixedmark />
-          </CheckboxBackground>
-        </CheckboxRoot>
-      );
+export class Checkbox extends withFoundation({
+  constructor: MDCCheckbox,
+  adapter: {}
+})<CheckboxPropsT> {
+  static displayName = 'Checkbox';
+
+  constructor(props: CheckboxPropsT) {
+    super(props);
+    this.generatedId = randomId('checkbox');
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    this.ripple_ = this.initRipple_();
+  }
 
-      /**
-       * We have to conditionally wrap our checkbox in a formfield
-       * If we have a label
-       */
-      if (label.length || children) {
-        return (
-          <FormField>
-            {checkbox}
-            <CheckboxLabel id={labelId + 'label'} htmlFor={labelId}>
-              {label}
-              {children}
-            </CheckboxLabel>
-          </FormField>
-        );
-      } else {
-        return checkbox;
-      }
+  syncWithProps(nextProps: CheckboxPropsT) {
+    // checked
+    syncFoundationProp(
+      nextProps.checked,
+      this.checked,
+      () => (this.checked = nextProps.checked)
+    );
+
+    // indeterminate
+    syncFoundationProp(
+      nextProps.indeterminate,
+      this.indeterminate,
+      () => (this.indeterminate = nextProps.indeterminate)
+    );
+
+    // disabled
+    syncFoundationProp(
+      nextProps.disabled,
+      this.disabled,
+      () => (this.disabled = nextProps.disabled)
+    );
+
+    // value
+    syncFoundationProp(
+      nextProps.value,
+      this.value,
+      () => (this.value = nextProps.value)
+    );
+  }
+
+  render() {
+    const {
+      label = '',
+      id,
+      children,
+      checked,
+      indeterminate,
+      apiRef,
```

Comment:
Does it make sense to still include `apiRef`? e.g. in the select component, we got rid of it.  (Same question for radio button)

Reply:
It actually still exists for now since I'm not sure if anyone was using it. See this line https://github.com/jamesmfriedman/rmwc/blob/mdc35/src/Base/MDCFoundation.js#L181

I didn't know this about React, but it turns out using a "ref" on a class component and not a Dom element actually gives you a reference to the class. So for all Foundation based components since they are merged with the original vanilla JS class, doing something like this would identical:

```
<Checkbox apiRef={api => {}} />
<Checkbox ref={api => {}} />
```
The only difference is timing. The standard React ref will fire during render where as the apiRef will only fire once the component mounts. As soon as all of the components are moved over to the new pattern, we can add a deprecation notice in the code and remove.

Reply:
Ah okay, actually that's what I did for all my selects that I am using: switching from `apiRef` to `ref`, cause on select the console says it does not recognize `apiRef`. 
But guess a big breaking change in the end for all components is the way to go

Reply:
Which means on select I forgot to remove it :)

Reply:
Thanks for the review @jodo and @amaesato